core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/store/impl/FileFactoryImplUnitTest.java

```diff
@@ -147,5 +149,39 @@ public static void tearDown() {
     FileFactory.getDataOutputStream(filePath, FileFactory.FileType.VIEWFS);
     assertNotNull(FileFactory.getCarbonFile(filePath, FileFactory.FileType.HDFS));
   }
+
+  @Test public void testTruncateFile() {
+    FileWriter writer = null;
+    try {
+      // generate a file
+      String path = new File("truncatFile").getCanonicalPath();
+      writer = new FileWriter(path);
+      for (int i = 0; i < 4000; i++) {
+        writer.write("test truncate file method");
+      }
+      writer.close();
+      CarbonFile file = FileFactory.getCarbonFile(path);
+      assertTrue(file.getSize() == 100000L);
+
+      // truncate file to 4000 bytes
+      FileFactory.truncateFile(
+          path,
+          FileFactory.getFileType(path),
+          4000);
+      file = FileFactory.getCarbonFile(path);
+      assertTrue(file.getSize() == 4000L);
```

Comment:
better to use assertEquals so that it will print the size

Reply:
fixed